domenetjenester/beregning-ytelse/src/main/java/no/nav/ung/sak/ytelse/RapportertInntektMapper.java

```diff
@@ -1,70 +1,215 @@
+
+        overlappendeArbeidsinntekter
+            .stream()
+            .map(it -> finnBeløpInnenforPeriode(intervall, it))
+            .reduce(BigDecimal::add)
+            .map(it -> new RapportertInntekt(InntektType.ARBEIDSTAKER_ELLER_FRILANSER, it))
+            .ifPresent(inntekterForPeriode::add);
+
+        final var overlappendeYtelse = finnOverlappendeInntekterForType(intervall, grupperteInntekter, InntektType.YTELSE);
+        overlappendeYtelse
+            .stream()
+            .map(it -> finnBeløpInnenforPeriode(intervall, it))
+            .reduce(BigDecimal::add)
+            .map(it -> new RapportertInntekt(InntektType.YTELSE, it))
+            .ifPresent(inntekterForPeriode::add);
+
+
+        final var tidslinjeForPeriode = new LocalDateTimeline<Set<RapportertInntekt>>(intervall, inntekterForPeriode);
+        return tidslinjeForPeriode;
+    }
+
+    private static void validerBrukersRapporterteInntekterTidslinje(LocalDateTimeline<Set<RapportertInntekt>> brukersRapporterteInntekter, LocalDateTimeline<Boolean> ytelseTidslinje) {
+        final var perioderUtenforYtelsesperioder = brukersRapporterteInntekter.getLocalDateIntervals().stream()
+            .filter(it -> ytelseTidslinje.getLocalDateIntervals().stream().noneMatch(it::equals))
+            .toList();
+
+        if (!perioderUtenforYtelsesperioder.isEmpty()) {
+            throw new IllegalStateException("Fant rapporterte inntekter for perioder utenfor ytelsesperioder: " + perioderUtenforYtelsesperioder);
+
+        }
+    }
+
+    private static LocalDateTimeline<RapporterteInntekter> kombinerTidslinjer(LocalDateTimeline<Set<RapportertInntekt>> brukersRapporterteInntekter, LocalDateTimeline<Set<RapportertInntekt>> registerTidslinje) {
+        return brukersRapporterteInntekter.crossJoin(registerTidslinje, (di, bruker, register) -> {
+            final var rapporterteInntekter = new RapporterteInntekter(bruker == null ? Set.of() : bruker.getValue(), register == null ? Set.of() : register.getValue());
+            return new LocalDateSegment<>(di, rapporterteInntekter);
+        });
+    }
+
+    private static BigDecimal finnBeløpInnenforPeriode(LocalDateInterval intervall, Inntektspost it) {
+        final var inntektsperiode = it.getPeriode();
+        final var overlapp = new LocalDateTimeline<>(intervall, true).intersection(new LocalDateInterval(inntektsperiode.getFomDato(), inntektsperiode.getTomDato()));
+        final var overlappPeriode = overlapp.getLocalDateIntervals().getFirst();
+
+        final var antallVirkedager = inntektsperiode.antallArbeidsdager();
+        final var overlappAntallVirkedager = Virkedager.beregnAntallVirkedager(overlappPeriode.getFomDato(), overlappPeriode.getTomDato());
+
+        return it.getBeløp().getVerdi().multiply(BigDecimal.valueOf(overlappAntallVirkedager).divide(BigDecimal.valueOf(antallVirkedager), 10, RoundingMode.HALF_UP));
```

Comment:
Er dette for å smøre månedsinntekten på antall dager i ytelsesperiode hvis det er forskjellig fra inntektsperiode?

Reply:
Ja, eg trur dei fleste arbeidsgivere registerer pr måned, men dersom det finnes inntekter som går over fleire måneder så beregner vi "bidraget" avhengig av kor stor del som overlapper med måneden vi er interessert i